### DISCUSSION TITLE
ILayer.Style and IFeature.Styles

### DISCUSSION THREAD

**Discussion by charlenni** [Mapsui/Mapsui#2415] [Development]
There is a difference in naming (singular and plural) and the types are different (ILayer.Style is IStyle? and IFeature.Styles is ICollection<IStyle>). Shouldn't be both the same (naming and type)?

**Comment by pauldendulk** [comment]
Hey, I just mentioned this in another issue. And I have mentioned this in the past in other issues. 

Advantages:
- ILayer and IFeature would be the same.
- It is clearer to users that they can add more than one style (it is possible to add multiple styles but users have to know that there is a StyleCollection).
- We do not have to make an arbitrary choice for a default style.
- The current default style is one of the most common confusions users have. There is an item for it in the FAQ: https://mapsui.com/documentation/faq.html#how-can-i-get-rid-of-the-white-dots-or-black-lines

So, that looks good, lots of advantages. But there are complications we need to take care of.
- If there is no default symbol users will not see anything in their map unless they explicitly add it. One solution would be to just leave this up to the user, but perhaps this will result in a lot of frustration. I don't have a clear picture of how that will work out.  Solutions:
  - Perhaps we should write a warning to the log, but then again sometimes the users just does not want a symbol (yes there will be cases where users want this), should we write a warning? 
  - Render something in a gray semi-transparent default style as long as there is no style. The user will at least see something. Perhaps only in debug mode.
  - Add a default symbol anyway. So, the layer will just have a single style when created.
- The current RasterFeature needs a RasterStyle. This is necessary in our logic because the style determines the renderer. It makes no sense for the user to add this, it adds no information, it is just necessary to make it work. Currently we solve this by setting a default style. This solution will disappear, so we have to come up with an alternative.
- Also every sample would need an extra line to add the style. It would be nice of the minimal code would be necessary to add a layer.

How does this work for other map controls?

Again, this is not an issue of just doing, but primarily of thinking.

**Comment by pauldendulk** [comment]
I now think we should go for it. I made this 'Enhancement' issue: https://github.com/Mapsui/Mapsui/issues/2452
